keras_hub/src/models/gemma/gemma_causal_lm.py

```diff
@@ -196,22 +196,102 @@ def call_with_cache(
             the final hidden representation of the input tokens, and `cache` is
             the decoding cache.
         """
-        x = self.backbone.token_embedding(token_ids)
-        x = x * ops.cast(ops.sqrt(self.backbone.hidden_dim), x.dtype)
-        # Each decoder layer has a cache; we update them separately.
-        caches = []
-        for i, transformer_layer in enumerate(self.backbone.transformer_layers):
-            current_cache = cache[:, i, ...]
-            x, next_cache = transformer_layer(
-                x,
-                cache=current_cache,
-                cache_update_index=cache_update_index,
+
+        use_openvino = keras.config.backend() == "openvino"
```

Comment:
this seems quite hard to maintain and like something we should probably avoid. why do we need a pass that is so different than other backends? shouldn't the point of keras' multi-backend setup allow to just call layers normally?